fourm/pseudolabel_frames.py

```diff
@@ -2,46 +2,88 @@
+os.makedirs(JSON_OUTPUT_DIR, exist_ok=True)
 
 # Load the YOLO model
-model = YOLO('bbox-yolo/yolov8n.pt')  # pretrained YOLOv8n model
+model = YOLO('/cluster/work/cotterell/yemara/ml-4m/bbox-yolo/yolov8n.pt') # pretrained YOLOv8n model
 
 # Extract the tar file
 with tarfile.open(SHARDS, "r") as tar:
-    tar.extractall(path="bbox-yolo/extracted_files")
+    tar.extractall(path="extracted_files")
 
 # Iterate through extracted files
-for root, dirs, files in os.walk("bbox-yolo/extracted_files"):
+for root, dirs, files in os.walk("extracted_files"):
     for file in files:
         if file.endswith(".mp4"):
             video_path = os.path.join(root, file)
             video = cv2.VideoCapture(video_path)
-            
             frame_paths = []
-            for frame_count in range(3):  # Only process the first 3 frames to test on euler
+            frame_count = 0
+            processed_frames = 0
+
+            while True:
                 success, frame = video.read()
                 if not success:
                     break
                 
-                # Save the frame as an image
-                frame_path = os.path.join(OUTPUT_DIR, f"{file[:-4]}_frame_{frame_count}.jpg")
-                cv2.imwrite(frame_path, frame)
-                frame_paths.append(frame_path)
-            
+                if frame_count % NTH_FRAME == 0:
+                    # Save the frame as an image
+                    frame_path = os.path.join(OUTPUT_DIR, f"{file[:-4]}_frame_{frame_count}.jpg")
+                    cv2.imwrite(frame_path, frame)
+                    frame_paths.append(frame_path)
+                    processed_frames += 1
+                
+                frame_count += 1
+
+                if MAX_FRAMES and processed_frames >= MAX_FRAMES:
+                    break
+
             video.release()
             
             # Apply pseudolabeling to the extracted frames
             results = model(frame_paths, project=LABELED_OUTPUT_DIR, name=file[:-4])
             
             for i, result in enumerate(results):
+                # Save labeled image
                 result.save(filename=f'{file[:-4]}_labeled_frame_{i}.jpg')
+                
+                # Extract bounding box information
+                boxes = result.boxes
+                json_data = []
+                for box in boxes:
+                    xyxy = box.xyxy[0].tolist()  # get box coordinates
+                    conf = box.conf.item()  # get confidence score
+                    cls = int(box.cls.item())  # get class id
+                    json_data.append({
+                        "bbox": xyxy,
+                        "confidence": conf,
+                        "class": cls,
+                        "class_name": result.names[cls]
+                    })
+                
+                # Save JSON file
+                json_filename = os.path.join(JSON_OUTPUT_DIR, f"{file[:-4]}_frame_{i}_boxes.json")
```

Comment:
Can we actually do 2 more things for saving the results?
1) aggregate the results in a list of jsons and save it using jsonl? (https://jsonlines.readthedocs.io/en/latest/) So, each video should be saved with the same name as the mp4 (so if a file is named 00004.mp4, it should be saved as 00004.jsonl).
2) repackage the jsonl's back into the tar files corresponding in name to the tarfiles containing those mp4s. For example, if all videos extracted from the tar video_rgb/00043.tar, should have its corresponding jsonls in video_det/00043.tar. See  https://github.com/swiss-ai/ml-4m/issues/11#issue-2391075201 for more deets!